napari/layers/shapes/_shapes_models/_polygon_base.py

```diff
@@ -90,39 +95,41 @@ def data(self, data):
         data = self.data_displayed
 
-        # splprep fails if two adjacent values are identical, which happens
-        # when a point was just created and the new potential point is set to exactly the same
-        # to prevent issues, we remove the extra points.
-        duplicates = np.isclose(data, np.roll(data, 1, axis=0))
-        # cannot index with bools directly (flattens by design)
-        data_spline = data[~np.all(duplicates, axis=1)]
-
-        if (
-            self.interpolation_order > 1
-            and len(data_spline) > self.interpolation_order
-        ):
-            data = data_spline.copy()
-            if self._closed:
-                data = np.append(data, data[:1], axis=0)
-
-            tck, *_ = splprep(
-                data.T, s=0, k=self.interpolation_order, per=self._closed
-            )
+        # # splprep fails if two adjacent values are identical, which happens
+        # # when a point was just created and the new potential point is set to exactly the same
+        # # to prevent issues, we remove the extra points.
+        # duplicates = np.isclose(data, np.roll(data, 1, axis=0))
+        # # cannot index with bools directly (flattens by design)
+        # data_spline = data[~np.all(duplicates, axis=1)]
+
+        if self.interpolation_order > 1:
+            data_spline = remove_path_duplicates(data, closed=True)
+
+            if len(data_spline) > self.interpolation_order:
+                data = data_spline.copy()
+                if self._closed:
+                    data = np.append(data, data[:1], axis=0)
 
-            # the number of sampled data points might need to be carefully thought
-            # about (might need to change with image scale?)
-            u = np.linspace(0, 1, self.interpolation_sampling * len(data))
+                tck, *_ = splprep(
+                    data.T, s=0, k=self.interpolation_order, per=self._closed
+                )
+
+                # the number of sampled data points might need to be carefully thought
+                # about (might need to change with image scale?)
+                u = np.linspace(0, 1, self.interpolation_sampling * len(data))
 
-            # get interpolated data (discard last element which is a copy)
-            data = np.stack(splev(u, tck), axis=1)[:-1]
+                # get interpolated data (discard last element which is a copy)
+                data = np.stack(splev(u, tck), axis=1)[:-1].astype(np.float32)
 
         # For path connect every all data
         self._set_meshes(data, face=self._filled, closed=self._closed)
-        self._box = create_box(self.data_displayed)
+        bbox = self._bounding_box[:, self.dims_displayed]
+        self._box = create_box_from_bounding(bbox)
 
-        self.slice_key = np.round(
-            self._bounding_box[:, self.dims_not_displayed]
-        ).astype('int')
+        self.slice_key = self._bounding_box[:, self.dims_not_displayed].astype(
+            'int'
+        )
```

Comment:
why is rounding removed here?

Reply:
It has significant performance impact (two numpy array allocations), and I was thinking that there is no change. But there is a change. 

But when investigating I have found https://numpy.org/doc/stable//reference/generated/numpy.rint.html#numpy.rint

I will update the code to use rint.